osbenchmark/benchmark.py

```diff
@@ -858,10 +858,10 @@ def dispatch_sub_command(arg_parser, args, cfg):
             cfg.add(config.Scope.applicationOverride, "builder", "preserve.install", convert.to_bool(args.preserve_install))
             cfg.add(config.Scope.applicationOverride, "system", "install.id", args.installation_id)
             builder.stop(cfg)
-        elif sub_command == "execute_test":
-            # As the execute_test command is doing more work than necessary at the moment, we duplicate several parameters
+        elif sub_command == "execute-test":
```

Comment:
Would make this change, but please don't drop support for `execute_test` for a couple of releases
```
elif sub_command == "execute-test" or sub_command == "execute_test":
```

Reply:
This change isn't enough to allow both `execute_test` and `execute-test` to coexist since `subparsers.add_parser()` currently takes in [a single title parameter](https://docs.python.org/3/library/argparse.html#sub-commands) (see below snippet of code), meaning it can only take in `execute_test` or `execute-test`. 

https://github.com/opensearch-project/opensearch-benchmark/blob/02d9b72f32ce6ca40357d51eb00e6be6868bc8b6/osbenchmark/benchmark.py#L114

To allow both to subcommand formats to coexist, we need to create a separate subparser for `execute-test` and add all the same arguments that are included under `execute_test` subparser. However, doing this will be cumbersome and would require duplicating code. Let me know if there's an easier work around that I'm not seeing.

Instead, is there a way we can publicly warn users ahead of time that `execute_test `will be deprecated by GA Release (1.0.0) or  --  worse case scenario  -- by a minor version after GA Release? Although we're only a couple weeks from GA release, I still think it'd be best to introduce `execute-test` by GA release regardless of whether we still support the legacy format `execute_test` or not.

Reply:
Those are good points.  There's no trivial workaround, and it's not worth the effort to maintain both variations.  We can note this in the documentation when that's released.  It would be appropriate to make the change in a version prior to the GA release, so there's some continuity.